src/unitxt/metrics.py

```diff
@@ -1524,7 +1529,7 @@ class Spearmanr(HuggingfaceMetric):
     hf_metric_name = "spearmanr"
     main_score = "spearmanr"
     process_single_instances = False
-    prediction_type = "float"
+    prediction_type = "Union[float, str]"
```

Comment:
Why should Spearman metric accept a string?
Any string to float conversion should be done in a post processor  in the template.

Reply:
All the metrics we use currently accept strings. We used the metric pipeline that does the conversion to float but it was removed in a recent commit by someone else. 
The best way to go is indeed to change all these metrics to accept floats and change the shared postprocessors accordingly - but this requires to change the api of other metrics in a non backward compatible way. Is that ok?
The relevant metrics: 
```
        "metrics.kendalltau_b",
        "metrics.roc_auc",
        "metrics.f1_binary",
        "metrics.accuracy_binary",
        "metrics.precision_binary",
        "metrics.recall_binary",
        "metrics.max_f1_binary",
        "metrics.max_accuracy_binary",
```

Reply:
This is the right thing to do.  I think most of these were added by you for kpa.   We will fix any remaining issues.

Reply:
Ok, changed the prediction_type to of the *_binary metrics to Union[float, int], since for binary labels it makes since to allow sending 0/1).